Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -197,6 +197,85 @@ u^2s & u^3 & t \cr
 structure VariableChange (R : Type u) [CommRing R] where
   (u : Rˣ) (r s t : R)
 
+namespace VariableChange
+
+section Group
+
+variable {R : Type u} [CommRing R]
+
+def one : VariableChange R where
+  u := 1
+  r := 0
+  s := 0
+  t := 0
+
+def mul (C C' : VariableChange R) : VariableChange R where
+  u := C.u * C'.u
+  r := C.r * ↑C'.u ^ 2 + C'.r
+  s := ↑C'.u * C.s + C'.s
+  t := C.t * ↑C'.u ^ 3 + C.r * C'.s * ↑C'.u ^ 2 + C'.t
+
+def inv (C : VariableChange R) : VariableChange R where
```

Comment:
You want docstrings on `def`s. They're not checked in the CI linter currently for the porting but you can check them locally with `#lint`, and I think they'll add this back eventually.